File: SlowQuitAppsLauncher/SQALAppDelegate.m
#import "SQALAppDelegate.h"

@implementation SQALAppDelegate

- (void)applicationDidFinishLaunching:(NSNotification *)_ {
    BOOL alreadyRunning = NO;
    for (NSRunningApplication *app in NSWorkspace.sharedWorkspace.runningApplications) {
        if ([app.bundleIdentifier isEqualToString:@"com.dteoh.SlowQuitApps"]) {
            alreadyRunning = YES;
            break;
        }
    }

    if (!alreadyRunning) {
        NSString *path = NSBundle.mainBundle.bundlePath;
        path = [path stringByDeletingLastPathComponent];
        path = [path stringByDeletingLastPathComponent];
        path = [path stringByDeletingLastPathComponent];
        path = [path stringByDeletingLastPathComponent];
        [NSWorkspace.sharedWorkspace launchApplication:path];
    }

    [NSApp terminate:self];
}

@end

File: SlowQuitAppsLauncher/SQALAppDelegate.swift
import Cocoa

@NSApplicationMain
class AppDelegate: NSObject, NSApplicationDelegate {
    func applicationDidFinishLaunching(_ aNotification: Notification) {
        let alreadyRunning = NSWorkspace.shared.runningApplications.contains {
            $0.bundleIdentifier == "com.dteoh.SlowQuitApps"
        }
        if !alreadyRunning {
            var path = Bundle.main.bundlePath as NSString
            for _ in 1...4 {
                path = path.deletingLastPathComponent as NSString
            }
            NSWorkspace.shared.launchApplication(path as String)
        }
    }
}
